### ISSUE TITLE
Suggestion: show road surface

### ISSUE THREAD

**Issue by joostschouppe** [der-stefan/OpenTopoMap#36]
Your map style looks like it would be feasible to show paved/unpaved roads without causing too much problems. Something similar to how tunnels are currently rendered, but without the faded colour might work. Similar to how this is done on old school Michelin style road maps, sometimes used for roads under construction. Especially when zoomed out this is a useful feature no other map has. Great test data is available in Argentina and Chile.

**Comment by der-stefan** [comment]
We show a mixture of surface, tracktype and the importance. So what would you like to improve concretely?

**Comment by joostschouppe** [comment]
I think you're talking about tracks here? I'm thinking about countries with less developed road networks, where having an unpaved primary (or even a trunk) road is nothing special. [Example in Bolivia](http://overpass-turbo.eu/s/h09). I don't see anything indicating surfacing (or lack thereof) in the opentopomap [there](https://opentopomap.org/#map=14/-16.19922/-67.73432).

**Comment by der-stefan** [comment]
You are right, I didn't think about roads. But that makes sense of course. We will consider this after switching to CartoCSS that makes it way easier.

**Comment by der-stefan** [comment]
There was no switch to CartoCSS yet, so nothing happened here.

**Comment by der-stefan** [comment]
We won't switch to CartoCSS, since the hand-made style is still alright.

**Comment by julcnx** [comment]
Coming back to this after checking the current mapnik style: it looks like the underlying mechanism for this already exists, just scoped to tracks and paths.

In [roads-casing.xml](https://github.com/der-stefan/OpenTopoMap/blob/master/mapnik/styles-otm/roads-casing.xml) and [roads-fill.xml](https://github.com/der-stefan/OpenTopoMap/blob/master/mapnik/styles-otm/roads-fill.xml), `highway=track` already gets a solid-to-dashed casing based on `tracktype` (grade1 solid through grade5 most dashed), and `highway=path`/`footway`/`bridleway` get the same treatment keyed on `surface=asphalt` vs everything else. Both `surface` and `tracktype` are already imported columns in [opentopomap.style](https://github.com/der-stefan/OpenTopoMap/blob/master/mapnik/osm2pgsql/opentopomap.style), so no schema change would be needed.

What's missing is applying that same dash logic to the standard road classes (`primary`, `secondary`, `tertiary`, `unclassified`, `residential`, `service`), which is what [@joostschouppe's Bolivia example](https://github.com/der-stefan/OpenTopoMap/issues/36#issuecomment-228675155) was pointing at.

One suggestion on the tagging side: `surface` rather than `smoothness` should drive this. `smoothness` encodes current rideability/condition (subjective, weather-dependent, re-tagged often), not material, so a rough paved road could render as "unpaved" and a well-graded gravel road as "paved". It's also far less used in the data: locally I count roughly 78M `surface` uses in OSM globally versus about 6.7M for `smoothness`, so keying the style on `surface` gives much more reliable coverage.

Happy to put together a PR extending the existing tracktype-style dashing to these road classes using `surface`, if that would be useful.

**Comment by julcnx** [comment]
Circling back on this. I offered a few weeks ago to put together a PR extending the existing tracktype-based dashing (already used for highway=track and path/footway/bridleway) to the standard road classes using surface, which is what the original Bolivia/Argentina examples in this thread were pointing at. Happy to go ahead and open that PR if the general direction sounds right, or to adjust the approach first if you'd rather it work differently.

**Comment by joostschouppe** [comment]
From the news about this project that I've seen, it seems unlikely that any work will be done on it at all (except for keeping it online)
